### PR TITLE
Prevent failures of Ansible task for service_enabled and timer_enabled

### PR DESCRIPTION
**Pull request by jan-cerny**
#### Description:
If the package which provides the service or timer is not present the Ansible task that enables the service or timer will fail because it can not find the unit file which will cause termination of the whole playbook. We can prevent the failure by executing the task only if the given package is present. The code uses Ansible package_facts module to determine if package is present.

#### Rationale:

Partially fixes: https://bugzilla.redhat.com/show_bug.cgi?id=1741455

Addressing:
```
TASK [Enable service fapolicyd] ************************************************
fatal: [192.168.122.24]: FAILED! => {"changed": false, "msg": "Could not find the requested service fapolicyd: host"}
```

### CODE REVIEW

File: shared/templates/template_ANSIBLE_service_enabled [hunk 1]
```diff
@@ -4,8 +4,18 @@
...
-    enabled: "yes"
-    state: "started"
+  block:
+  - name: Check if {{{ PACKAGENAME }}} is installed
+    package:
+      name: {{{ PACKAGENAME }}}
+      state: present
+    check_mode: true
+    register: pkg
+  - name: Enable service {{{ SERVICENAME }}}
```
Comment (redhatrises) [inline shared/templates/template_ANSIBLE_service_enabled]: Add a blank line for readability please.

**Comment by redhatrises** [comment]
I wonder if it would be better to run `package_facts.packages[my_package_name]` instead of the `package` module.

**Comment by jan-cerny** [comment]
@openscap-ci test this please

File: shared/templates/template_ANSIBLE_timer_enabled [hunk 2]
```diff
@@ -4,8 +4,15 @@
...
 - name: Enable timer {{{ TIMERNAME }}}
-  systemd:
-    name: "{{{ TIMERNAME }}}.timer"
-    enabled: "yes"
-    state: "started"
+  block:
+  - name: Gather the package facts
+    package_facts:
+      manager: auto
+  - name: Enable timer {{{ TIMERNAME }}}
```
Comment (redhatrises) [inline shared/templates/template_ANSIBLE_timer_enabled]: Need a blank line before `-name` here as well

File: shared/templates/template_ANSIBLE_timer_enabled [hunk 3]
```diff
@@ -4,8 +4,15 @@
...
+    systemd:
+      name: "{{{ TIMERNAME }}}.timer"
+      enabled: "yes"
+      state: "started"
+    when: |-
+      '{{{ PACKAGENAME }}}' in ansible_facts.packages
```
Comment (redhatrises) [inline shared/templates/template_ANSIBLE_timer_enabled]: Let's make this:
```
   when:
     - '{{{ PACKAGENAME }}}' in ansible_facts.packages
```

File: shared/templates/template_ANSIBLE_service_enabled [hunk 4]
```diff
@@ -4,8 +4,15 @@
...
+    package_facts:
+      manager: auto
 
+  - name: Enable service {{{ SERVICENAME }}}
+    service:
+      name: "{{{ DAEMONNAME }}}"
+      enabled: "yes"
+      state: "started"
+    when: |-
+      '{{{ PACKAGENAME }}}' in ansible_facts.packages
```
Comment (redhatrises) [inline shared/templates/template_ANSIBLE_service_enabled]: Let's make this:
```
   when:
     - '{{{ PACKAGENAME }}}' in ansible_facts.packages
```

File: shared/templates/template_ANSIBLE_timer_enabled (hunk 2 above)
Comment (jan-cerny) [reply shared/templates/template_ANSIBLE_timer_enabled]: fixed

File: shared/templates/template_ANSIBLE_timer_enabled (hunk 3 above)
Comment (jan-cerny) [reply shared/templates/template_ANSIBLE_timer_enabled]: fixed

File: shared/templates/template_ANSIBLE_service_enabled (hunk 4 above)
Comment (jan-cerny) [reply shared/templates/template_ANSIBLE_service_enabled]: fixed

**Comment by yuumasato** [comment]
LGTM, but every task for service enabled will be checking for the list of packages installed.
Similar to what happens here:
https://github.com/ComplianceAsCode/content/blob/3147bfe95b4154de447f92b82c44e222ea6d4995/shared/templates/template_ANSIBLE_sebool_var#L8

I'm afraid there is no easy way to avoid this, unless generation of Ansible playbooks is smart enough to de-duplicate tasks. And the block may actually make it harder to de-duplicate.

**Comment by redhatrises** [comment]
> LGTM, but every task for service enabled will be checking for the list of packages installed.
> Similar to what happens here:
> 
> https://github.com/ComplianceAsCode/content/blob/3147bfe95b4154de447f92b82c44e222ea6d4995/shared/templates/template_ANSIBLE_sebool_var#L8
> 
> I'm afraid there is no easy way to avoid this, unless generation of Ansible playbooks is smart enough to de-duplicate tasks. And the block may actually make it harder to de-duplicate.

@yuumasato this is more of the correct way of checking if a package is installed rather than what is in the sebool_var. You will always have to run package_facts more than once if validating that a package is installed.

**Comment by jan-cerny** [comment]
> generation of Ansible playbooks is smart enough to de-duplicate tasks

We build that using `oscap xccdf generate fix` so it would need to greatly improve this part of `oscap` preferably in a  way that it understands YAML

**Pull request merged**